app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleViewHolders.kt

```diff
@@ -21,15 +21,226 @@ class LocaleViewHolder(
+            "ast" to "Asturianu",
+            "az" to "Azerbaijani",
+            "be" to "Belarusian",
+            "bg" to "Bulgarian",
+            "bn" to "Bengali",
+            "br" to "Breton",
+            "bs" to "Bosnian",
+            "ca" to "Catalan",
+            "cak" to "Kaqchikel",
+            "ceb" to "Cebuano",
+            "co" to "Corsican",
+            "cs" to "Czech",
+            "cy" to "Welsh",
+            "da" to "Danish",
+            "de" to "German",
+            "dsb" to "Sorbian, Lower",
+            "el" to "Greek",
+            "eo" to "Esperanto",
+            "es" to "Spanish",
+            "et" to "Estonian",
+            "eu" to "Basque",
+            "fa" to "Persian",
+            "ff" to "Fulah",
+            "fi" to "Finnish",
+            "fr" to "French",
+            "fy-NL" to "Frisian",
+            "ga-IE" to "Irish",
+            "gd" to "Gaelic",
+            "gl" to "Galician",
+            "gn" to "Guarani",
+            "gu-IN" to "Gujarati",
+            "he" to "Hebrew",
+            "hi-IN" to "Hindi",
+            "hil" to "Hiligaynon",
+            "hr" to "Croatian",
+            "hsb" to "Sorbian, Upper",
+            "hu" to "Hungarian",
+            "hy-AM" to "Armenian",
+            "id" to "Indonesian",
+            "is" to "Icelandic",
+            "it" to "Italian",
+            "ja" to "Japanese",
+            "ka" to "Georgian",
+            "kab" to "Kabyle ",
+            "kk" to "Kazakh",
+            "kmr" to "Kurmanji Kurdish",
+            "kn" to "Kannada",
+            "ko" to "Korean",
+            "lij" to "Ligurian",
+            "lo" to "Lao",
+            "lt" to "Lithuanian",
+            "mix" to "Mixtepec Mixtec",
+            "ml" to "Malayalam",
+            "mr" to "Marathi",
+            "ms" to "Malay",
+            "my" to "Burmese",
+            "nb-NO" to "Norwegian Bokmål",
+            "ne-NP" to "Nepali ",
+            "nl" to "Dutch, Flemish",
+            "nn-NO" to "Norwegian Nynorsk",
+            "nv" to "Navajo, Navaho",
+            "oc" to "Occitan",
+            "pa-IN" to "Punjabi",
+            "pl" to "Polish",
+            "pt-BR" to "",
+            "pt-PT" to "",
+            "rm" to "Romansh",
+            "ro" to "Română",
+            "ru" to "Russian",
+            "sat" to "Santali",
+            "sk" to "Slovak",
+            "sl" to "Slovenian",
+            "sq" to "Albanian",
+            "sr" to "Serbian",
+            "su" to "Sundanese",
+            "sv-SE" to "Swedish",
+            "ta" to "Tamil",
+            "te" to "Telugu",
+            "tg" to "Tajik",
+            "th" to "Thai",
+            "tl" to "Tagalog",
+            "tr" to "Turkish",
+            "trs" to "Triqui",
+            "uk" to "Ukrainian",
+            "ur" to "Urdu",
+            "uz" to "Uzbek",
+            "vec" to "Venitian",
+            "vi" to "Vietnamese"
+        )
+    }
 }
 
 class SystemLocaleViewHolder(
```

Comment:
It seems reasonable to me that we would want smaller classes and files in general for easier readability and maintenance. I would also want to consider the opportunity to extract `SystemLocaleViewHolder` and `BaseLocaleViewHolder` into their own files.

Reply:
@gabrielluong I agree, but this is some technical debt existing in many other files and some of them are a matter of style and choice (i.e. having an interface in the same file or different file than classes that implement it). I would gladly handle this tech debt issue in a separate PR.